src/app/content/components/Page/highlightManager.ts

```diff
@@ -110,6 +110,25 @@ export default (container: HTMLElement, getProp: () => HighlightProp) => {
     }
   };
 
+  const insertPendingCardInOrder = (highlights: Highlight[], pending: Highlight) => {
+    if (!highlighter) { return highlights; }
+
+    const ordered = highlights.filter((highlight) => !pending || highlight.id !== pending.id);
+    const prevHighlight = highlighter.getHighlightBefore(pending);
+    if (!prevHighlight) {
+      ordered.unshift(pending);
+      return ordered;
+    }
+
+    const indexToInsert = ordered.findIndex((highlight) => highlight.id === prevHighlight.id);
```

Comment:
if you change this condition to be like this you can delete the conditional above
```suggestion
    const indexToInsert = ordered.findIndex((highlight) => prevHighlight && highlight.id === prevHighlight.id);
```